lib/queue.js

```diff
@@ -771,7 +758,13 @@ Queue.prototype.processJob = function(job){
 */
 Queue.prototype.getNextJob = function(opts){
   if(!this.closing){
-    return this.moveJob('wait', 'active', opts).then(this.getJobFromId);
+    var _this = this;
+
+    return ((this.rateLimiter && this.rateLimiter()) || Promise.resolve())
```

Comment:
we could also assign Promise.resolve to this.rateLimiter when no rateLimiter defined,  for a simpler code at this point.